src/geanyobject.c

```diff
@@ -156,6 +156,12 @@ static void create_signals(GObjectClass *g_object_class)
 		G_SIGNAL_RUN_FIRST,
 		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
 		G_TYPE_NONE, 0);
+	geany_object_signals[GDB_PROJECT_BEFORE_CLOSE] = g_signal_new (
+		"project-before-close",
+		G_OBJECT_CLASS_TYPE (g_object_class),
+		G_SIGNAL_RUN_FIRST,
+		0, NULL, NULL, g_cclosure_marshal_VOID__VOID,
+		G_TYPE_NONE, 0);
```

Comment:
Hum, actually I just thought of a perfectly plausible use case: get notified the settings were changed back to the non-project one and read those.  This requires the signal to be emitted after those were restored, and that currently mean when the project is actually closed.

So maybe we need that extra signal.